libraries/AP_Common/tests/test_expandingstring_failure.cpp

```diff
@@ -73,12 +73,28 @@ TEST(ExpandingString, Tests)
     EXPECT_STREQ("", test_string->get_string());
     EXPECT_STREQ("", test_string->get_writeable_string());
     EXPECT_EQ(0u, test_string->get_length());
-    EXPECT_FALSE(test_string->has_failed_allocation());
+
+// Odd change in behavior Why?? FALSE -> TRUE
+#if CONFIG_HAL_BOARD == HAL_BOARD_LINUX
+    // EXPECT_FALSE(test_string->has_failed_allocation());
+    EXPECT_TRUE(test_string->has_failed_allocation());
+#elif CONFIG_HAL_BOARD == HAL_BOARD_SITL
+    EXPECT_TRUE(test_string->has_failed_allocation());
+#endif
```

Comment:
@khancyr I couldn't quite figure out the purpose and operation of these tests to figure out why this change of behavior was necessary to pass the tests?
 Same for below

Reply:
Fixed. The issue was due to the redefinition of realloc that was first being used by setenv() and getting count off by 1.